ckanext/scheming/fanstatic/scheming-repeating-subfields.js

```diff
@@ -23,10 +23,17 @@ this.ckan.module('scheming-repeating-subfields', function (jQuery, _) {
 
             $remove.on('click', function(e) {
                 var $groups = $this.find('.scheming-subfield-group'),
-                    $last = $groups.last(),
+                    $curr = $(this).closest('.scheming-subfield-group'),
+                    index = $curr.data('groupIndex'),
+                    field = $curr.data('field'),
                     count = $groups.length;
                 if(count !== 1) {
-                    $last.remove();
+                    document.getElementById("statusText").innerHTML = "Are you sure you want to remove "+field+" "+(index+1)+"?";
+                    $("#confirmRemoval").on('click', function(e) {
+                        $curr.remove();
+                    });
+                }else{
+                    document.getElementById("statusText").innerHTML = "Can not remove, at least one "+field+" must be present";
```

Comment:
is the remove button not part of the repeating element? In what case would this be triggered?

Reply:
The remove button is repeated with every sub -field. When one is clicked, it finds the closest parent sub-field with `$curr = $curr = $(this).closest('.scheming-subfield-group'),`

Reply:
I see. How about allowing removal of the last one but create a new blank record when that happens?

Reply:
That's a good idea, I'll add it